test/StaticAnalysis/AllowListChecks.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LaminasTest\Filter\StaticAnalysis;
+
+use Laminas\Filter\AllowList;
+
+/** @psalm-suppress UnusedClass */
+final class AllowListChecks
+{
+    public function filterReturnTypeIsUnionOfInputAndNull(int $value): int|null
```

Comment:
since there is `null` allowed here, I guess `null` works in the template?
so even if you remove `int` from return value here, that static analysis will still pass, right?
Not sure if that is how it supposed to be?

Reply:
This test verifies that when input type is known, it will be present in the returned union. Without the `int` here, the union would be `mixed|null`

Reply:
but for the allowlist there is `null` defined while it is actually the value which is returned when the filtering was not successful? even tho, the return type should be a templated type from constructor, but I guess that is not possible to declare 🤔 since return type would be value-of list option.

imho, this Filter should not exist as it combines validation with filtering to a different value. the value is converted to sth invalid when stuff is not as it should be. I would expect s1 from combining validator rather than depending on filtering. but might be a personal thing